### PR TITLE
[BitSerializer] Update to v0.50

### LINKED ISSUE #28572: [BitSerializer] update to 0.50
Library name: BitSerializer

New version number: 0.50

Other information that may be useful (release notes, etc...)

Release notes: https://bitbucket.org/Pavel_Kisliak/bitserializer/src/dev/History.md

Added new component "csv-archive" which links statically (all other parts is "header-only").

PR: #28564

### PR DESCRIPTION
**Pull request by PavelKisliak**
**Describe the pull request**

- #### What does your PR fix?
  Updates BitSerializer to version 0.50

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  `all (but only with static linkage)`, `No`

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  `Yes`

### CODE REVIEW

File: ports/bitserializer/portfile.cmake [hunk 1]
```diff
@@ -1,8 +1,13 @@
+# All components of BitSerializer is "header only" except CSV archive
+if (${BUILD_CSV_ARCHIVE})
+    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+endif()
```
Comment (dg0yt) [inline ports/bitserializer/portfile.cmake:4]: Flipping linkage by feature doesn't look right. If everything else is header-only, why use `if` at all?

Comment (PavelKisliak) [reply ports/bitserializer/portfile.cmake]: I was also doubt how will be right.
Thanks, fixed.

Comment (dg0yt) [reply ports/bitserializer/portfile.cmake]: On second sight, what's the problem with the CSV lib? If it is only DLL symbol export, dynamic linkage might work well for systems other than windows, and the statement might be wrapped into `if(VCPKG_TARGET_IS_WINDOWS)`. This pattern exists in some other ports.

Comment (PavelKisliak) [reply ports/bitserializer/portfile.cmake]: In addition to export symbols on Windows, library doesn't have CMake code which support build shared library.
This is not big work, but as author, I don't see reasons to support shared linkage, the library is quite small.
If you know any cases where shared library will be useful even the library size is small, please let me know.

**Review by JonLiu1993 — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Comment by JonLiu1993** [comment]
Tested usage successfully with `BitSerializer:x64-windows` triplet

**Pull request merged**